TACC/Core-CMS
# feat: bump core-styles & use its django-cms-forms

## Status

- [x] [Build CMS image.](https://jenkins01.tacc.utexas.edu/job/Core_CMS_Build/1206/console)
- [x] Resolve [v3.12.0-alpha.11 blockers](https://github.com/TACC/Core-CMS/releases/tag/v3.12.0-alpha.11) on APCD and Frontera.
- [x] Deploy on Frontera (Pre-Prod).
- ~~Deploy on APCD (Staging).~~
    <sub>APCD is can not use v4 until it is [migrated to v3.12](https://github.com/TACC/Core-CMS-Custom/blob/main/docs/port-project.md#migrate-project-to-core-cms-v312).</sub>
- [x] **Deploy on TACC (Dev).**
- [x] Screenshot.

## Overview

[TACC/Core-Styles@v2.15.0](https://github.com/TACC/Core-Styles/releases/tag/v2.15.0) loads `django-cms-forms.css` (in `core-styles.base.css` via `c-form`)[^1], so do not duplicate it.

## Related

- builds atop #742
- similar to https://github.com/TACC/tup-ui/pull/279

## Changes

- remove `django-cms-forms` styles
- load core-styles version that provides `djangocms-cms-forms`[^1]

[^1]: Because it seems easier to support via Core-Styles than in isolation in Core-CMS. Because Core-Styles uses `c-form` for `s-form` and `djaango-cms-forms` in [a DRY manner](https://github.com/TACC/Core-Styles/blob/main/src/lib/_imports/tools/selectors.form.css) that I only see becoming redundant if maintained in Core-CMS. Questions and ideas welcome. Form CSS is easy to refactor now.

## Testing

1. Deploy on Frontera (Pre-Prod) and APCD (Staging) and TACC (Dev).
2. Verify the following forms look as they did before.
    - https://dev.tup.tacc.utexas.edu/about/help/
    - https://pprd.frontera-portal.tacc.utexas.edu/test/tup-230-form-plugin/
    - https://pprd.frontera-portal.tacc.utexas.edu/test/tup-320-form-plugin/
    - https://pprd.frontera-portal.tacc.utexas.edu/test/sal-test-form/
    - ~~https://apcd-qa.tacc.utexas.edu/register/request-to-submit/~~
    - ~~https://apcd-qa.tacc.utexas.edu/submissions/threshold-exception/~~
    - ~~https://apcd-qa.tacc.utexas.edu/submissions/extension-request/~~

## UI

### TACC

Uses Django CMS Form plugin.

| Form | Before | After |
| - | - | - |
| [Help](https://dev.tup.tacc.utexas.edu/about/help/) | ![before](https://github.com/TACC/Core-CMS/assets/62723358/44b2f590-b611-4dd7-956a-69672dc6005c) | ![after](https://github.com/TACC/Core-CMS/assets/62723358/562adf7a-2340-45d7-91e1-3d90347a72fd) |

### Frontera

Uses Django CMS Form plugin.

| Form | Before | After |
| - | - | - |
| [TUP-230](https://pprd.frontera-portal.tacc.utexas.edu/test/tup-230-form-plugin/) | ![tup-230](https://github.com/TACC/Core-CMS/assets/62723358/31101703-e65a-4583-bca0-7e99b4fba6d9) | ![230](https://github.com/TACC/Core-CMS/assets/62723358/e499e872-3380-4ccf-adb2-601f8c96ec2f)
| [TUP-320](https://pprd.frontera-portal.tacc.utexas.edu/test/tup-320-form-plugin/) | ![tup-320](https://github.com/TACC/Core-CMS/assets/62723358/83d1c9d9-95e2-47c7-a493-08666298df66) | ![320](https://github.com/TACC/Core-CMS/assets/62723358/38afc826-d0d1-4bba-9be9-08cbb43db218)
| [Sal Test Form](https://pprd.frontera-portal.tacc.utexas.edu/test/sal-test-form/) | ![sal test form](https://github.com/TACC/Core-CMS/assets/62723358/fc712c1e-ff0d-48ea-83de-e02ec8cdb41a) | ![sal](https://github.com/TACC/Core-CMS/assets/62723358/74040293-6257-409d-9190-da244d873f95)

### APCD

<details><summary>Untested</summary>

Uses Django CMS Form markup. [Sad story.](https://confluence.tacc.utexas.edu/x/vJd9E)

| Form | Before | After |
| - | - | - |
| [Request to Submit](https://apcd-qa.tacc.utexas.edu/register/request-to-submit/) | ![request to submit](https://github.com/TACC/Core-CMS/assets/62723358/cca76b34-b0d3-4fbc-8f16-85a0bc696e83) | …
| [Threshold Exception](https://apcd-qa.tacc.utexas.edu/submissions/threshold-exception/) | ![threshold exception](https://github.com/TACC/Core-CMS/assets/62723358/92a94668-745e-491c-b823-46785a3e70bc) | …
| [Extension Request](https://apcd-qa.tacc.utexas.edu/submissions/extension-request/) | ![extension request](https://github.com/TACC/Core-CMS/assets/62723358/690ded7d-df5b-486e-a384-2c8e170af765) | …

</details>

## Notes

Core-Styes v12–14 do not include any breaking changes.
